pkg/pool-utils/contracts/RecoveryMode.sol

```diff
@@ -60,7 +60,7 @@ abstract contract RecoveryMode is IRecoveryMode, BasePoolAuthorization {
      * @dev Does not otherwise affect pool operations (beyond deferring payment of protocol fees), though some pools may
      * perform certain operations in a "safer" manner that is less likely to fail, in an attempt to keep the pool
      * running, even in a pathological state. Unlike the Pause operation, which is only available during a short window
-     * after factory deployment, Recovery Mode can always be enableed.
+     * after factory deployment, Recovery Mode can always be enabled.
      */
     function enableRecoveryMode() external override authenticate {
         _setRecoveryMode(true);
```

Comment:
Neither `enableRecoveryMode` or `disableRecoveryMode` check the current recovery mode state prior to calling `_setRecoveryMode`.  It doesn't impact the `BasePool` implementation, but logically I'd expect there to be some sort of guard to prevent setting recovery mode to a pool already in recovery mode, and vice-a-versa.

Reply:
I don't think there's a good way that we can fully enforce this without considering the Authorizer and permissions as well.

If we had this guard then you could still cause problems by "flicking" the pool in and out of recovery mode whenever you want to trigger the hook (assuming you had permissions for both functions). We'd then need to bake in cooldowns (or similar mechanism) which would be impossible to perfectly set from the outset. This sort of thing probably makes more sense to push out to the periphery and rely on a helper contract or delays on disabling recovery mode.